misc/log4j/src/main/java/org/openscience/cdk/tools/LoggingTool.java

```diff
@@ -448,6 +448,17 @@ public void warn(Object object, Object... objects) {
     public boolean isDebugEnabled() {
         return doDebug;
     }
+    
+    /**
+     * Enables debugging output. 
+     * Use this method for computational demanding debug info.
+     * 
+     * @see isDebugEnabled()
+     */
+    @Override
+    public void setDebugEnabled(boolean enabled) {
```

Comment:
I think we may want to throw an exception here, to indicate people are not supposed to use this method on the Log4J implementation, with a proper comment/pointer to how to configure it properly.

What do you think?

Reply:
Or just set the Log4J to debug...?

Reply:
@johnmay, that's sort of the point... people should set Log4J to debug, and not use the Java API... one solution.

Reply:
The whole point was I wanted to use SystemOutLoggingTool and that, just as LoggingTool, extends ILoggingTool. SystemOutLoggingTool would *never* log *anything* (it is independet from log4j) because it has setDebugEnabled() method. I agee that it is good to have LoggingTool throw an exception if you use setDebugEnabled. Or remove SystemOutLogging tool for the sake of consistency, as you can use the console logger of log4j in the LoggingTool. Hope that makes sense?!

Reply:
I think it makes sense to have SystemOutLLogging have debug=true by default... otherwise, just don't register it... the only reason (and I think that's exactly it is why that method exists) is the use in end tools, where you want to be able to turn this on and of... think JChemPaint (RIP), Bioclipse, KNIME, rcdk, etc, etc...